### PR TITLE
feat: [CDS-79068]: Implement new deploy API and add clean option

### PR DESCRIPTION
**Pull request by abosii-harness**
## Describe your changes
As part of this PR we're moving from the old API to a new deploy API to support clean option. As per now, this API is enabled only when the clean option is enabled, but can be easy changed in the future

**Tested cases**
- [x] [Linux][Jar] Successful slot deployment
- [x] [Linux][War] Successful slot deployment
- [x] [Linux][Zip/Dotnet App] Successful slot deployment
- [x] [Windows][Jar] Successful slot deployment
- [x] [Windows][War] Successful slot deployment
- [x] [Windows][Zip/Dotnet App] Successful slot deployment

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/53048)

### CODE REVIEW

**Comment by abosii-harness** [comment]
trigger checkstylecodeformat

**Comment by abosii-harness** [comment]
trigger uts

File: 960-api-services/src/main/java/io/harness/azure/impl/AzureWebClientImpl.java [hunk 1]
```diff
@@ -874,6 +881,36 @@ public Mono<Void> deployZipToSlotAsync(AzureWebClientContext context, final Stri
...
+  }
+
+  @Override
+  public Mono<Void> deployToWebAppAsync(
+      AzureWebClientContext context, DeployType type, File file, DeployOptions options) {
+    if (file == null) {
+      throw new IllegalArgumentException(FILE_BLANK_ERROR_MSG);
+    }
+
+    log.debug("Start deploying zip file async, fileAbsoluteFile: {}, context: {}", file.getAbsolutePath(), context);
```
Comment (markoharness) [inline 960-api-services/src/main/java/io/harness/azure/impl/AzureWebClientImpl.java:909]: Its not zip always, right? Shouldn't we pull this info based on deploy type?

**Review by alebuh — APPROVED** [review]

File: 960-api-services/src/main/java/io/harness/azure/impl/AzureWebClientImpl.java (hunk 1 above)
Comment (abosii-harness) [reply 960-api-services/src/main/java/io/harness/azure/impl/AzureWebClientImpl.java]: yeah, you're right. Just a copy paste mistake, fixed now

**Comment by abosii-harness** [comment]
trigger checkstylecodeformat

**Comment by abosii-harness** [comment]
trigger uts

**Review by markoharness — APPROVED** [review]

**Review by alebuh — APPROVED** [review]

**Review by GoelPratyush99 — APPROVED** [review]

**Comment by abosii-harness** [comment]
trigger ut9

**Comment by abosii-harness** [comment]
trigger ut9

**Comment by abosii-harness** [comment]
trigger review

**Pull request merged**